tests/functional/database-management.test.js

```diff
@@ -9,13 +9,107 @@ describe('database management', function () {
+                    createDB: false,
+                    configOptions: {owner_public_key: harnessConfigs.masterPublicKey},
+                    private_pem: harnessConfigs.masterPrivateKey,
+                    public_pem: harnessConfigs.masterPublicKey
+                }));
+
+            context('with matching client private key and daemon owner_public_key', function () {
+
+                it('should be able to successfully createDB', async function () {
+                    await this.api._createDB()
+                });
+
+                it('should be able to successfully updateDB', async function () {
+                    await this.api._updateDB(5000)
+                });
+
+                it('should be able to successfully deleteDB', async function () {
+                    await this.api._deleteDB();
+                });
+            });
+
+            context('with mismatching client private key and daemon owner_public_key', function () {
+
+                before('create "no access" client', async function () {
+
+                    this.esrAddress = harnessConfigs.testRemoteSwarm ? harnessConfigs.esrContractAddress : this.swarmManager.getEsrContractAddress();
+
+                    const apis = await initializeClient({
+                        createDB: false,
+                        uuid: randomUuid,
+                        esrContractAddress: this.esrAddress,
+                        private_pem: notMasterKeyPair.privateKey,
+                        public_pem: notMasterKeyPair.publicKey
+                    });
+
+                    this.noAccessApi = apis[0];
+                });
+
+                it('should fail to createDB', async function () {
+                    await this.noAccessApi._createDB().should.be.rejectedWith('ACCESS_DENIED')
+                });
+
+                it('should fail to updateDB', async function () {
+                    await this.noAccessApi._updateDB().should.be.rejectedWith('ACCESS_DENIED')
+                });
+
+                it('should fail to deleteDB', async function () {
+                    await this.noAccessApi._deleteDB().should.be.rejectedWith('ACCESS_DENIED')
+                });
+
+                context('with an existing database', function () {
+
+                    before('createDB with owner client', async function () {
+                        await this.api._createDB();
+                    });
+
+                    it('should fail to createDB', async function () {
+                        await this.noAccessApi._createDB().should.be.rejectedWith('ACCESS_DENIED')
+                    });
+
+                    it('should fail to updateDB', async function () {
+                        await this.noAccessApi._updateDB().should.be.rejectedWith('ACCESS_DENIED')
+                    });
+
+                    it('should fail to deleteDB', async function () {
+                        await this.noAccessApi._deleteDB().should.be.rejectedWith('ACCESS_DENIED')
```

Comment:
I would especially like a behavioural test here that the DB was not deleted, e.g. try to create a key and assert that creation is still possible.

Reply:
There shouldn't be a DB here, but I've added two behavioural tests after the createDB and deleteDB calls to ensure that even with the master key, a create key operation fails with `DATABASE_NOT_FOUND`